common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -57,3 +64,165 @@ def perform_destroy(self, instance):
+        CourseEntitlement.set_enrollment(entitlement, None)
+
+    def create(self, request, uuid):
+        """
+        On POST this method will be called and will handle enrolling a user in the
+        provided course_run_id from the data. This is called on a specific entitlement
+        UUID so the course_run_id has to correspond to the Course that is assigned to
+        the Entitlement.
+
+        When this API is called for a user who is already enrolled in a run that User
+        will be unenrolled from their current run and enrolled in the new run if it is
+        available.
+        """
+        course_run_id = request.data.get('course_run_id', None)
+
+        if not course_run_id:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data='The Course Run ID was not provided.'
+            )
+
+        # Verify that the user has an Entitlement for the provided Course UUID.
+        try:
+            entitlement = CourseEntitlement.objects.get(uuid=uuid, user=request.user, expired_at=None)
+        except CourseEntitlement.DoesNotExist:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data='The Entitlement for this UUID does not exist or is Expired.'
+            )
+
+        # Verify the course run ID is of the same type as the Course entitlement.
+        course_run_valid = self._verify_course_run_for_entitlement(entitlement, course_run_id)
+        if not course_run_valid:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'message': 'The Course Run ID is not a match for this Course Entitlement.'
+                }
+            )
+
+        # Determine if this is a Switch session or a simple enroll and handle both.
+        try:
+            course_run_string = CourseKey.from_string(course_run_id)
+        except InvalidKeyError:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'message': 'Invalid {course_id}'.format(course_id=course_run_id)
+                }
+            )
+        if entitlement.enrollment_course_run is None:
+            self._enroll_entitlement(
+                entitlement=entitlement,
+                course_run_key=course_run_string,
+                user=request.user
+            )
+        elif entitlement.enrollment_course_run.course_id != course_run_id:
+            self._unenroll_entitlement(
+                entitlement=entitlement,
+                course_run_key=entitlement.enrollment_course_run.course_id,
+                user=request.user
+            )
+            self._enroll_entitlement(
+                entitlement=entitlement,
+                course_run_key=course_run_string,
+                user=request.user
+            )
+
+        return Response(
+            status=status.HTTP_201_CREATED,
+            data={
```

Comment:
I'm fine with this.  But can we put a comment above noting that the serializer is not used.

Can we get away with not setting a serializer if we don't use it?